### PR TITLE
Add doc about Shoot status

### PR DESCRIPTION
**Pull request by ialidzhikov**
**How to categorize this PR?**

/area documentation
/kind task
/priority normal

**What this PR does / why we need it**:
Add documentation about the Shoot status.

**Release note**:

```improvement user
A [documentation for the Shoot status](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_status.md) has been added.
```

### CODE REVIEW

**Review by rfranzke — APPROVED** [review]
/lgtm

**Review by timebertt — APPROVED** [review]
Very nice doc! Thanks for picking this up ❤️ 
/lgtm

**Comment by timebertt** [comment]
I think this doc also deserves a release note to make end users aware of it (some already asked for such documentation), don't you think 😉

**Comment by ialidzhikov** [comment]
> Very nice doc! Thanks for picking this up ❤️
> /lgtm

Thank you for pointing out "kind/documentation" task. I now added a relnote. :)

File: docs/usage/shoot_status.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+##### Error Codes
+
+Known error codes are:
+- `ERR_INFRA_UNAUTHORIZED` - indicates that the last error occurred due to invalid infrastructure credentials
+- `ERR_INFRA_INSUFFICIENT_PRIVILEGES` - indicates that the last error occurred due to insufficient infrastructure privileges
+- `ERR_INFRA_QUOTA_EXCEEDED` - indicates that the last error occurred due to infrastructure quota limits
+- `ERR_INFRA_DEPENDENCIES` - indicates that the last error occurred due to dependent objects on the infrastructure level
+- `ERR_INFRA_RESOURCES_DEPLETED` - indicates that the last error occurred due to depleted resource in the infrastructure
+- `ERR_CLEANUP_CLUSTER_RESOURCES` - indicates that the last error occurred due to resources in the cluster that are stuck in deletion
+- `ERR_CONFIGURATION_PROBLEM` - indicates that the last error occurred due a configuration problem
```
Comment (timuthy) [inline docs/usage/shoot_status.md:76]:
```suggestion
- `ERR_CONFIGURATION_PROBLEM` - indicates that the last error occurred due to a configuration problem
```

File: docs/usage/shoot_status.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+### Conditions
+
+The Shoot status consists of a set of conditions. A [Condition](https://gardener.cloud/documentation/references/core/#core.gardener.cloud/v1beta1.Condition) has the following fields:
+
+| Field name           | Description                                                                                                        |
+| -------------------- | ------------------------------------------------------------------------------------------------------------------ |
+| `type`               | Name of the condition.                                                                                             |
+| `status`             | Indicates whether the condition is applicable, with possible values `True`, `False`, `Unknown`, or `Progressing`.  |
+| `lastTransitionTime` | Timestamp for when the condition last transitioned from one status to another.                                     |
+| `lastUpdateTime`     | Timestamp for when the condition was updated.                                                                      |
```
Comment (timuthy) [inline docs/usage/shoot_status.md:14]:
```suggestion
| `lastUpdateTime`     | Timestamp for when the condition was updated. Usually changes when `reason` or `message` in condition is updated                                                                     |
```

**Review by timuthy — COMMENTED** [review]
Just two comments, although already merged. Thanks for the nice document!

**Pull request merged**